src/components/AddTodoForm.jsx

```diff
@@ -1,37 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
+// Third party libraries:
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 const AddTodoForm = ({onAddTodo}) => {
+  /*
+    ============================
+    =          STATES          =
+    ============================
+  */
+  const [todoTitle, setTodoTitle] = useState("");
+
   /*
     ============================
     =         HANDLERS         =
     ============================
   */
+  const handleTitleChange = ({target}) => {
+    const newTodoTitle = target.value;
+    setTodoTitle(newTodoTitle);
+  }
 
   const handleFormSubmit = (event) => {
     event.preventDefault();
-    const todoTitle = event.target.title.value.trim();
     /*
       Validation:
         - If todoTitle submitted is blank, then do not change the state.
         Else, change the state.
     */
-    if(!todoTitle){
+    if(!todoTitle.trim()){
       return;
     }
-    console.log(todoTitle);
-    onAddTodo(todoTitle);
-    event.target.reset();
+    onAddTodo({
+      title: todoTitle.trim(),
+      id: Date.now()
+    });
+    setTodoTitle("");
   };
 
   return (
-    <form onSubmit={handleFormSubmit}>
-        <label htmlFor="todoTitle">Title: </label>
-        <input id="todoTitle" type="text" name="title"/>
-        <button type="submit">Add</button>
-    </form>
+          <form onSubmit={handleFormSubmit}>
+            <label htmlFor="todoTitle">
+            <FontAwesomeIcon
+                icon={["fas", "calendar-plus"]}
+                size="xl"
+              />
+              <span>Title:</span>
+            </label>
+            <input 
+              id="todoTitle" 
+              type="text" 
+              placeholder='Add a task'
+              value={todoTitle} 
+              onChange={handleTitleChange}
+              name="title"
+            />
+            <button type="submit">Add</button>
+          </form>
   );
 };
 
 export {
     AddTodoForm as default
 };
+
+/*
+  ============================
+  =        PROP TYPES        =
+  ============================
+*/
+AddTodoForm.propTypes = {
+  onAddTodo: PropTypes.func.isRequired
+};
```

Comment:
I thought the propTypes were way ahead in the book haha! :grin:

Reply:
> I thought the propTypes were way ahead in the book haha! 😁

Really? I wanted to see if I could validate react component's prop and I eventually found out about propTypes which was pretty useful to what I wanted haha, I did not know it was covered in the book way ahead

Reply:
Have you take a look at Typescript yet?

Reply:
I haven't yet! Based on the name I suppose it is a type of JS focused on variable types, isn't it? 🤔